### PR TITLE
use: ZIO Fibers directly to reduce interop overhead

### PR DESCRIPTION
**Pull request by hearnadam**

### CODE REVIEW

File: kyo-zio/shared/src/main/scala/kyo/ZIOs.scala [hunk 1]
```diff
@@ -67,4 +73,28 @@ object ZIOs:
         }
     end run
 
+    private object internal:
+        extension [E, A](exit: zio.Exit[E, A])
```
Comment (fwbrasil) [inline kyo-zio/shared/src/main/scala/kyo/ZIOs.scala:77]: Could you introduce tests for these extensions? I wonder if we should make them public, I imagine they could be useful in user code as well

File: kyo-zio/shared/src/main/scala/zio/FiberInterop.scala [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package zio
```
Comment (fwbrasil) [inline kyo-zio/shared/src/main/scala/zio/FiberInterop.scala:1]: What do you think about creating an issue in ZIO to provide this API? I'm ok with the workaround but we could migrate to a proper API if the issue gets addressed

File: kyo-zio/shared/src/main/scala/kyo/ZIOs.scala [hunk 3]
```diff
@@ -17,15 +19,19 @@ object ZIOs:
       * @return
       *   A Kyo effect that, when run, will execute the zio.ZIO
       */
-    def get[E, A](v: => ZIO[Any, E, A])(using Frame, zio.Trace): A < (Abort[E] & Async) =
+    def get[E, A](v: => ZIO[Any, E, A])(using f: Frame, t: zio.Trace): A < (Abort[E] & Async) =
```
Comment (fwbrasil) [inline kyo-zio/shared/src/main/scala/kyo/ZIOs.scala:22]: it seems naming the implicits isn't necessary

File: kyo-zio/shared/src/main/scala/zio/FiberInterop.scala (hunk 2 above)
Comment (hearnadam) [reply kyo-zio/shared/src/main/scala/zio/FiberInterop.scala:1]: I will open a PR later this week.

Comment (hearnadam) [reply kyo-zio/shared/src/main/scala/zio/FiberInterop.scala:1]: https://github.com/zio/zio/pull/9246

**Pull request merged**